templates/Dockerfile-rpm.tmpl

```diff
@@ -40,6 +40,7 @@ RUN if [[ %%os%%   == oraclelinux ]] && [[ %%release%% == 7 ]]; then yum install
 # install build tools and PostgreSQL development files
 RUN ( yum install -y https://%%rpm_url%% \
     && [[ -z "%%extra-repositories%%" ]] || yum install -y %%extra-repositories%%) \
+    && yum update -y \
```

Comment:
We already have the same command in the first `RUN` on line 4:

```
RUN yum -y update
```

Why do we need to update all the packages one more time? Is this really necessary?

You may have issues if you use some caches for the intermediate steps. That can be solved by adding `--no-cache` option when using `docker build`

Reply:
We are adding new repositories so yum update is needed after adding new repo

Reply:
> yum update is needed after adding new repo

No it is not. We do not need to upgrade all the packages in the system after we install a new repo.

Do you see any `yum update` commands in our repo installation scripts?

Here is where we install the new repo: https://github.com/citusdata/packaging/blob/71e84ec5a2a038e770af45a1a53b65e1cd26e9f9/community/rpm.sh#L45

Reply:
I see your point. You're right it seems that yum install <repo_name> updates caches but still there are some non-up to date packages.  I can not find any other alternatives other thant yum update. Yum check-update only check for updates if there is shows the results. So I understand that first I need to call yum check-update and if there are updates I will call yum update. But yum update performs these steps internally and if there is no update it just print that all packages are up-to-date. So solely calling yum update makes more sense for me